frontend/packages/dev-console/src/components/import/devfile/DevfileImportForm.tsx

```diff
@@ -23,8 +29,59 @@ const DevfileImportForm: React.FC<FormikProps<FormikValues> & DevfileImportFormP
 }) => {
   const { t } = useTranslation();
   const [, devfileParseError] = useDevfileServer(values, setFieldValue);
+  const searchParams = new URLSearchParams(window.location.search);
+  const gitRepoUrl = searchParams.get('gitRepo');
+  const devfileName = searchParams.get('devfileName');
+  const formType = searchParams.get('formType');
+  const [devfileSamples, setDevfileSamples] = React.useState<DevfileSample[]>();
+  const selectedSample = React.useMemo(
+    () => devfileSamples?.find((sample) => sample.name === devfileName),
+    [devfileSamples, devfileName],
+  );
   useDevfileDirectoryWatcher(values, setFieldValue);
 
+  React.useEffect(() => {
```

Comment:
Would be probably good to move all this logic into a custom hook.

Reply:
Moved to custom hook.